src/alert/alert.spec.ts

```diff
@@ -21,79 +22,179 @@ describe('ngb-alert', () => {
+      expect(getCloseButton(alertEl)).toBeTruthy();
+
+      getCloseButton(alertEl).click();
+      fixture.detectChanges();
+      expect(getAlertElement(fixture.nativeElement)).toBeNull();
+    });
+
+    it('should open a dismissible alert with a specified type', () => {
+      const fixture = createTestComponent(`<template ngbAlert type="success">Hello, {{name}}!</template>`);
+      const alertEl = getAlertElement(fixture.nativeElement);
+
+      expect(alertEl).toHaveCssClass('alert-success');
+      expect(getCloseButton(alertEl)).toBeTruthy();
+    });
+
+    it('should dismiss alert and invoke close handler on close button click', () => {
+      const fixture = createTestComponent(`<template ngbAlert (close)="closed = true">Hello, {{name}}!</template>`);
+      const alertEl = getAlertElement(fixture.nativeElement);
 
-    getCloseButton(alertEl).click();
-    fixture.detectChanges();
+      getCloseButton(alertEl).click();
+      fixture.detectChanges();
 
-    expect(fixture.componentInstance.closed).toBeTruthy();
-    expect(getAlertElement(fixture.nativeElement)).toBeNull();
+      expect(fixture.componentInstance.closed).toBeTruthy();
+      expect(getAlertElement(fixture.nativeElement)).toBeNull();
+    });
+
+    it('should auto close on timeout specified', fakeAsync(() => {
+         const fixture =
+             createTestComponent(`<template ngbAlert [dismissOnTimeout]="1000">Hello, {{name}}!</template>`);
+         const alertEl = getAlertElement(fixture.nativeElement);
+
+         expect(alertEl.getAttribute('role')).toEqual('alert');
+         expect(getCloseButton(alertEl)).toBeTruthy();
+
+         tick(800);
+         fixture.detectChanges();
+         expect(alertEl.getAttribute('role')).toEqual('alert');
+         expect(getCloseButton(alertEl)).toBeTruthy();
+
+         tick(1200);
+         fixture.detectChanges();
+         expect(getAlertElement(fixture.nativeElement)).toBeNull();
+       }));
   });
 
-  it('should auto close on timeout specified', fakeAsync(() => {
-       const fixture = createTestComponent(`<template ngbAlert [dismissOnTimeout]="1000">Hello, {{name}}!</template>`);
-       const alertEl = getAlertElement(fixture.nativeElement);
+  describe('Custom config', () => {
 
-       expect(alertEl.getAttribute('role')).toEqual('alert');
-       expect(getCloseButton(alertEl)).toBeTruthy();
+    beforeEach(() => { TestBed.configureTestingModule({declarations: [TestComponent], imports: [NgbAlertModule]}); });
 
-       tick(800);
-       fixture.detectChanges();
-       expect(alertEl.getAttribute('role')).toEqual('alert');
-       expect(getCloseButton(alertEl)).toBeTruthy();
+    it('should initialize inputs with provided config', () => {
+      let config: NgbDismissibleAlertConfig;
+      const fixture = createTestComponent(`<template ngbAlert>Hello, {{name}}!</template>`);
+      inject([NgbDismissibleAlertConfig], (c: NgbDismissibleAlertConfig) => {
```

Comment:
The idea is to simulate what the config service documentation suggests to do to globally change the default values for alerts (and all the other components):

 - NgbAlertModule declares an NgbAlertConfig provider 
 - an application main module (simulated by the test module here) imports NgbAlertModule (or NgbModule). A global NgbAlertConfig service, containing mutable default values, is thus available to the whole application
 - the root application component gets a reference to this global NgbAlertConfig config thanks to dependency injection. That is simulated by the inject() in the test
 - the root application component sets custom default values in the injected NgbAlertConfig service. That is simulated by the body of the function passed to inject().
 - A template of the application ends up using an NgbAlert component. This is simulated by the createTestComponent() call in the test
 - The created alert should use the default values set before in the NgbAlertConfig service. This is what the test checks.

Does that clarify? I agree the test is more convoluted than the other similar tests, but checking for the right timeout can't be done by just inspecting the dom, and, to be fair, I haven't found a way to just getting an instance of the NgbDismissibleAlert in the test.

Using

    TestBed.createComponent(NgbDismissibleAlert)

leads to the following error message, which looks completely wrong to me:
    
    Cannot create the component NgbDismissibleAlert as it was not imported into the testing module!